### ISSUE TITLE
csd-power keeps inhibitor on suspending after unplugging laptop from monitor

### ISSUE THREAD

**Issue by TheShed412** [linuxmint/Cinnamon#7347]
```
OS: Mint 18.2 sonya
Kernel: x86_64 Linux 4.10.0-041000-generic
DE: Cinnamon 3.4.6
WM: Muffin
WM Theme: Vivaldi (Numix Daily)
Theme: Mint-Y-Dark [GTK2/3]
Icon Theme: Numix-Circle
Font: Noto Sans 9
CPU: Intel Core i5-7200U CPU @ 3.1GHz
GPU: Mesa DRI Intel(R) HD Graphics 620 (Kaby Lake GT2) 
RAM: 1098MiB / 7620MiB
```

**Issue**
After using an external monitor, csd-power keeps an inhibitor that will not allow my laptop to suspend.

**Steps to reproduce**

* Make sure that your laptop can properly suspend when closing the lid
* Keep it powered on when plugging it in to a monitor.
* Close lid.
* Open lid and unplug from monitor

**Expected behaviour**
* The suspend function will operate as it did before using the monitor

**Actual Behavior**
* Suspend only makes the computer idle, as if a monitor is still plugged in. 

**Other information**
List of blocking inhibitors. This was printed while the laptop was unplugged from the monitor after multiple reboots.

```
╰─$ systemd-inhibit --list --mode=block
     Who: tyler (UID 1000/tyler, PID 1598/csd-media-keys)
    What: handle-power-key:handle-suspend-key:handle-hibernate-key
     Why: Cinnamon handling keypresses
    Mode: block

     Who: tyler (UID 1000/tyler, PID 1596/csd-power)
    What: handle-lid-switch
     Why: Multiple displays attached
    Mode: block

2 inhibitors listed.

```
I have attempted to use logind.conf to try to remedy this but it never recognizes my changes even after multiple reboots. I have set the session manager to default to logind, but that doesn't work.
```
org.cinnamon.desktop.session session-manager-uses-logind true
org.cinnamon.desktop.session session-name 'cinnamon'
org.cinnamon.desktop.session settings-daemon-uses-logind true
org.cinnamon.desktop.session idle-delay uint32 600
org.cinnamon.desktop.session screensaver-uses-logind false

```
logind.conf:
```
[Login]
#NAutoVTs=6
#ReserveVT=6
#KillUserProcesses=no
#KillOnlyUsers=
#KillExcludeUsers=root
InhibitDelayMaxSec=5
HandlePowerKey=ignore
HandleSuspendKey=suspend
HandleHibernateKey=suspend
HandleLidSwitch=suspend
HandleLidSwitchDocked=suspend
#PowerKeyIgnoreInhibited=no
#SuspendKeyIgnoreInhibited=no
#HibernateKeyIgnoreInhibited=no
LidSwitchIgnoreInhibited=no

```
This does seem to be a problem with Cinnamon/csd-power. I have AWM and Mate install beside Cinnamon and neither have this issue. Here is a related post on the systemd github: 
https://github.com/systemd/systemd/issues/7137

**Comment by TheShed412** [comment]
I recently installed Linux Mint 18.3 on a live usb and tested out sleeping. It works fine. It even has the same inhibitors as my main install. So it isn't the inhibitor itself that is causing the issue. But it is definitely an issue with cinnamon because both mate and AWM sleep fine when the lid is shut.

**Comment by orestarod** [comment]
The result of `systemd-inhibit --list`:
  ```
[user@user-pc ~]$ systemd-inhibit --list  
     Who: user (UID 1000/user, PID 21706/csd-media-keys)
    What: handle-power-key:handle-suspend-key:handle-hibernate-key
     Why: Cinnamon handling keypresses
    Mode: block

     Who: ModemManager (UID 0/root, PID 655/ModemManager)
    What: sleep
     Why: ModemManager needs to reset devices
    Mode: delay

     Who: NetworkManager (UID 0/root, PID 662/NetworkManager)
    What: sleep
     Why: NetworkManager needs to turn off networks
    Mode: delay

     Who: user (UID 1000/user, PID 21698/csd-power)
    What: sleep
     Why: Cinnamon needs to lock the screen
    Mode: delay

     Who: user (UID 1000/user, PID 21698/csd-power)
    What: handle-lid-switch
     Why: Multiple displays attached
    Mode: block

     Who: UPower (UID 0/root, PID 926/upowerd)
    What: sleep
     Why: Pause device polling
    Mode: delay

6 inhibitors listed.
```

**Comment by orestarod** [comment]
The result of `systemd-inhibit --list`:
  ```bash
[user@user-pc ~]$ systemd-inhibit --list  
     Who: user (UID 1000/user, PID 21706/csd-media-keys)
    What: handle-power-key:handle-suspend-key:handle-hibernate-key
     Why: Cinnamon handling keypresses
    Mode: block

     Who: ModemManager (UID 0/root, PID 655/ModemManager)
    What: sleep
     Why: ModemManager needs to reset devices
    Mode: delay

     Who: NetworkManager (UID 0/root, PID 662/NetworkManager)
    What: sleep
     Why: NetworkManager needs to turn off networks
    Mode: delay

     Who: user (UID 1000/user, PID 21698/csd-power)
    What: sleep
     Why: Cinnamon needs to lock the screen
    Mode: delay

     Who: user (UID 1000/user, PID 21698/csd-power)
    What: handle-lid-switch
     Why: Multiple displays attached
    Mode: block

     Who: UPower (UID 0/root, PID 926/upowerd)
    What: sleep
     Why: Pause device polling
    Mode: delay

6 inhibitors listed.
```

**Comment by orestarod** [comment]
Regarding your problem, in System Settings > Power Management > Power , there is an option saying `Perform lid-closed action even with external monitors attached`, which is I believe by default set to false.

**Comment by TheShed412** [comment]
Yeah, I've set that to true. It seems to be something else. Regardless of what I set any options to it's ignored by Cinnamon. I'm not sure if the issue is here or the daemon over on the mint repo. 

Even if that was the issue, it's still thinking an external monitor is attached when one is not.

**Comment by TheShed412** [comment]
It also doesn't suspend properly with any other method. So it's not just the lid issue. It's just any suspend is being blocked by something.

**Comment by TheShed412** [comment]
It typically just idles and I can't wake it up after. The keyboard back light comes on but the screen never does. I'm going to just let it idle for a awhile like that to see if any time makes it suspend. As of what I've tested so far it hasn't suspended in under 5 minutes.

**Comment by TheShed412** [comment]
So I have updated to Linux Mint 18.3, the issue still persists. When the install finished, it suspended once properly, and hasn't since first updating.

**Comment by slano** [comment]
I have a very similar issue (lid/suspend behavior changed after unplugging external monitor).

My laptop is set to Do nothing on lid close when on A/C power. This works as expected, but only until I plug and unplug my external monitor. After unplugging the external monitor my laptop suspends on lid close while on A/C power. After restarting behavior returns back to normal.

**Comment by Ahmed-E-86** [comment]
> I have a very similar issue (lid/suspend behavior changed after unplugging external monitor).
> 
> My laptop is set to Do nothing on lid close when on A/C power. This works as expected, but only until I plug and unplug my external monitor. After unplugging the external monitor my laptop suspends on lid close while on A/C power. After restarting behavior returns back to normal.

 I have the same issue, but seeing that this issue is not fixed from 2018 does not encourage me to keep using Cinnamon.

**Comment by Ahmed-E-86** [comment]
> I have a very similar issue (lid/suspend behavior changed after unplugging external monitor).
> 
> My laptop is set to Do nothing on lid close when on A/C power. This works as expected, but only until I plug and unplug my external monitor. After unplugging the external monitor my laptop suspends on lid close while on A/C power. After restarting behavior returns back to normal.

I have the same issue, but after searching I found a way to force what happens when the lid is closed
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/desktop_migration_and_administration_guide/closing-lid
